Move method bindings from ArrayMesh to Mesh:

playmyskay opened the pull request:
1. surface_get_primitive_type
2. surface_get_format
3. surface_get_array_index_len
4. surface_get_array_len

I would extend cpp-bindings to get access from gdnative modules via api.json. I have already successfully tested this.
I would appreciate it if this would be included in the **3.2 branch**.

playmyskay committed:
Moved method bindings from ArrayMesh to Mesh:
1. surface_get_primitive_type
2. surface_get_format
3. surface_get_array_index_len
4. surface_get_array_len

clayjohn commented:
You'll need to also update the corresponding documentation by running the doctool command e.g. ``path_to_engine --doctool .`` and moving the descriptions over.

akien-mga commented on scene/resources/mesh.cpp:
```diff
@@ -467,6 +467,11 @@ Size2i Mesh::get_lightmap_size_hint() const {
 }
 
 void Mesh::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("surface_get_primitive_type", "surf_idx"), &Mesh::surface_get_primitive_type);
+	ClassDB::bind_method(D_METHOD("surface_get_format", "surf_idx"), &Mesh::surface_get_format);
+	ClassDB::bind_method(D_METHOD("surface_get_array_index_len", "surf_idx"), &Mesh::surface_get_array_index_len);
+	ClassDB::bind_method(D_METHOD("surface_get_array_len", "surf_idx"), &Mesh::surface_get_array_len);
+
```
I'd move them down with the other `surface_` method bindings.

akien-mga commented:
I'm not sure this can be cherry-picked for `3.2`, won't this break the GDNative ABI?
CC @BastiaanOlij

playmyskay committed:
Updated docs
Moved method bindings from ArrayMesh to Mesh

BastiaanOlij commented:
No, should be fine, as they are new bindings they just won’t be available
unless you regenerate the bindings. It does mean that if you use these but
load the plugin in an older version of godot it will break, but thats been
a problem with cpp bindings from day one.

On Mon, 31 Aug 2020 at 7:19 pm, Rémi Verschelde <notifications@github.com>
wrote:

> I'm not sure this can be cherry-picked for 3.2, won't this break the
> GDNative ABI?
> CC @BastiaanOlij <https://github.com/BastiaanOlij>
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/godotengine/godot/pull/41632#issuecomment-683667150>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAO262OB6Z7YJYWIY75HKH3SDNTLDANCNFSM4QP2LFHA>
> .
>
-- 
Kindest regards,

Bastiaan Olij

My personal profile on Facebook <https://www.facebook.com/bastiaan.olij>
My professional profile on LinkedIn
<https://www.linkedin.com/in/bastiaanolij/>
My profile on Twitter <https://twitter.com/mux213>
My tutorials on Youtube <https://www.youtube.com/BastiaanOlij>
My code repositories on Github <https://github.com/BastiaanOlij>

aaronfranke commented:
@playmyskay Is this still desired? If so, this needs to be rebased on the latest master branch, and squashed, along with addressing all of the above feedback.

playmyskay commented:
> @playmyskay Is this still desired? If so, this needs to be rebased on the latest master branch, and squashed, along with addressing all of the above feedback.

@aaronfranke 
No longer desired.

pwnorbitals commented:
@playmyskay should this issue be closed then ?
